juriscraper/pacer/appellate_docket.py

```diff
@@ -917,7 +920,16 @@ def _get_tail_by_regex(self, regex, cast_to_date=False, node=None):
         node = node if node is not None else self.tree
         nodes = node.re_xpath(f'//*[re:match(text(), "{regex}")]')
         try:
-            tail = clean_string(nodes[0].tail.strip())
+            tail_text = nodes[0].tail
+            tail_text = tail_text.strip() if tail_text is not None else ""
+            # If there's no tail text and we're parsing the docket_number,
+            # fall back to parsing the docket_number wrapped in a "tel:" href.
+            if not tail_text and regex == self.docket_number_lookup_regex:
+                a_nodes = nodes[0].xpath(
+                    "following-sibling::a[starts-with(@href, 'tel:')][1]"
+                )
+                tail_text = a_nodes[0].text or "" if a_nodes else ""
+            tail = clean_string(tail_text)
```

Comment:
This feels like it abuses this method to do specific things for docket numbers. Is there a better way that can keep this method general?

Reply:
Sure! I've left this method as it was and introduced the `_parse_docket_number` method, 
which contains the fallback logic to parse the `docket_number` inside a "tel:" href tag.